Add circuit breakers (#5055)
wso2/docs-apim: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/endpoints/resiliency/circuit-breakers.md

# Circuit Breakers

> In case of backend failure it is always better to fail quickly and relieve the back pressure from the actual backend, so it can recover quickly. The Choreo Connect supports various types of distributed circuit breakers offered by the Envoy. 

Reviewer: ```suggestion
In case of backend failure, it is always better to fail quickly and relieve the back pressure from the actual backend, so it can recover quickly. Choreo Connect supports various types of distributed circuit breakers offered by Envoy. 
```

Based on the circuit breaker configuration, the router (i.e. Envoy) will open the circuit and handle the consequent request by itself till the backend recovers. 

## Enable circuit breakers for API endpoints

!!! note

>     Circuit breakers are enabled by default and have modest default values.

Reviewer: ```suggestion
    Circuit breakers are enabled by default and have moderate default values.
```
Can we use moderate or something similar here?

Author: yes. that would be fine

> The following is the format of the vendor extension which requires for enabling circuit breaking for an endpoint cluster in Choreo Connect.

Reviewer: ```suggestion
The following is the format of the vendor extension that is required to enable circuit breaking for an endpoint cluster in Choreo Connect.
```
